bittide/src/Bittide/Transceiver.hs

```diff
@@ -0,0 +1,232 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Bittide.Transceiver (transceiverPrbs) where
+import qualified Clash.Prelude as C
+import Clash.Explicit.Prelude
+import Clash.Cores.Xilinx.GTH hiding (ChansUsed)
+import Clash.Xilinx.ClockGen
+import Clash.Annotations.Primitive    (hasBlackBox, Primitive (InlineYamlPrimitive), HDL(..))
+
+import Clash.Annotations.SynthesisAttributes
+import Data.String.Interpolate (__i)
+
+import Clash.Cores.Xilinx.VIO
+
+
+concatBvs :: (KnownNat n, KnownNat m) => Vec n (BitVector m) -> BitVector (n*m)
+concatBvs = bitCoerce
+
+splitBvs :: (KnownNat n, KnownNat m) => BitVector (n*m) -> Vec n (BitVector m)
+splitBvs = bitCoerce
+
+
+transceiverPrbs
```

Comment:
About the record idea, I feel that might limit or at least make it annoying to higher-order things with with `transceiverPrbs`, like in `transceiverPrbsN`:

https://github.com/bittide/bittide-hardware/blob/80a8f288950a6c4070c76b1a07189ac889870f25/bittide/src/Bittide/Transceiver.hs#L51-L52


Also I'm a little afraid to mess too much with the high-speed serial signals (in the `txS` and `rxS` domains), like packing them into and pulling them out of datastructures, because these signals aren't routable, they have fixed connections from the GTH transceivers to dedicated IO pins.
For that to work correctly Vivado has to be able to understand that all the intermediate code we generate and conclude that they're just straight "wires".
Now I don't actually how problematic this might be, I haven't see it go wrong yet, but I'm afraid it might when the intermediate code gets to complicated.

Reply:
Fair enough, let's ditch the record idea.